packages/playwright/src/takeSnapshot.ts

```diff
@@ -37,7 +40,11 @@ async function takeSnapshot(
     rrwebSnapshot.snapshot(document);
   `);
 
-  testInfo.attach(name, { contentType, body: JSON.stringify(domSnapshot) });
+  const bufferedSnapshot = Buffer.from(JSON.stringify(domSnapshot));
+  if (!chromaticSnapshots[testId]) {
+    chromaticSnapshots[testId] = {};
```

Comment:
We should use a `Map` here instead of an object since a map guarantees the key order based on insertion order so we can be sure snapshots will be in the correct order. Also, it will allow us to use `chromaticSnapshots.delete(testId)` rather than the `delete` operator used in `makeTest`.